src/CoinGeckoClient.ts

```diff
@@ -32,19 +32,31 @@ import {
  * The wrap client to access all api on coin gecko
  */
 export class CoinGeckoClient {
-  apiV3Url = "https://api.coingecko.com/api/v3";
+  private static readonly API_V3_URL = "https://api.coingecko.com/api/v3";
+
+  private static readonly PRO_API_V3_URL = "https://pro-api.coingecko.com/api/v3";
 
   options: Options = {
     timeout: 30000,
     autoRetry: true,
   };
 
+  baseURL: string;
+
+  apiKey?: string;
+
   /**
    * Constructor
    * @param options the options passed for client library, at the moment only timeout are support
    */
-  constructor(options?: Options) {
+  constructor(options?: Options, apiKey?: string) {
```

Comment:
I think about create new class and inherit from this class 
class CoinGeckoClient extend CoinGeckoClient() {
 constructor(apiKey: string) {
-- set the base URl
-- set default header 
}
}